conans/client/installer.py

```diff
@@ -541,19 +536,21 @@ def _call_package_info(self, conanfile, package_folder, ref, is_editable):
                 self._hook_manager.execute("pre_package_info", conanfile=conanfile,
                                            reference=ref)
                 if hasattr(conanfile, "layout"):
-                    # Old cpp info without defaults (the defaults are in the new one)
-                    conanfile.cpp_info = CppInfo(conanfile.name, package_folder,
-                                                 default_values=CppInfoDefaultValues())
                     if not is_editable:
                         package_cppinfo = conanfile.cpp.package.copy()
                         package_cppinfo.set_relative_base_folder(conanfile.folders.package)
-                        # Copy the infos.package into the old cppinfo
-                        fill_old_cppinfo(conanfile.cpp.package, conanfile.cpp_info)
+                        conanfile.cpp_info = package_cppinfo
                     else:
-                        conanfile.cpp_info.filter_empty = False
+                        conanfile.cpp_info = conanfile.cpp.package.copy()
+                        conanfile.cpp_info.clear()
```

Comment:
This is because you could previously declare a component attribute in the `layout` method in the conanfile and in package_info you are appending a path to it, as it was previously initialised. But in editable mode that information is discarded and you don't want to use that path but still you have an append to the attribute, so here we get the information that the component attribute was declared in the `cpp.package` but discard the path it was set to with `clear()`

Reply:
As discussed, all that part has been removed now. Now the cpp_info from the conanfile is the same as cpp.info and the old logic will probably be changed.